libsql-server/src/namespace/meta_store.rs

```diff
@@ -322,18 +374,40 @@ impl MetaStore {
         tracing::debug!("removing namespace `{}` from meta store", namespace);
 
         let mut guard = self.inner.lock();
-        guard.conn.execute(
-            "DELETE FROM namespace_configs WHERE namespace = ?",
-            [namespace.as_str()],
-        )?;
-        if let Some(sender) = guard.configs.remove(&namespace) {
+        let r = if let Some(sender) = guard.configs.get(&namespace) {
             tracing::debug!("removed namespace `{}` from meta store", namespace);
             let config = sender.borrow().clone();
+            let tx = guard.conn.transaction()?;
+            if let Some(shared_schema) = config.config.shared_schema_name.as_deref() {
+                tx.execute(
+                    "DELETE FROM shared_schema_links WHERE shared_schema_name = ? AND db_name = ?",
+                    [shared_schema, namespace.as_str()],
+                )?;
+            }
+            if config.config.is_shared_schema {
+                let res: rusqlite::Result<u32> = tx.query_row(
+                    "SELECT COUNT(*) FROM shared_schema_links WHERE shared_schema_name = ?",
+                    [namespace.as_str()],
+                    |row| row.get(0),
+                );
+                match res {
+                    Ok(0) | Err(rusqlite::Error::QueryReturnedNoRows) => { /* ok */ }
+                    Ok(n) => return Err(Error::SharedSchemaDestroyError(namespace.to_string(), n)),
+                    Err(e) => return Err(e.into()),
+                }
+            }
+            tx.execute(
+                "DELETE FROM namespace_configs WHERE namespace = ?",
```

Comment:
shouldn't we also delete the link?

Reply:
The link is removed above in lines 381-386.